app/src/common/chromium/com/igalia/wolvic/browser/api/impl/TabWebContentsDelegate.java

```diff
@@ -77,7 +86,20 @@ public void navigationStateChanged(int flags) {
     }
 
     @Override
-    public void onWebAppManifest(WebContents webContents, @NonNull String manifest) {}
+    public void onWebAppManifest(WebContents webContents, @NonNull String manifest) {
+         @Nullable WSession.ContentDelegate delegate = mSession.getContentDelegate();
+        if (delegate != null) {
```

Comment:
You can do the reverse check and do an early return to save some indentation

Reply:
Done